File: problems/3014-minimum-number-of-pushes-to-type-word-i/solution.py
class Solution:
    def minimumPushes(self, word: str) -> int:
        counts = {}
        for char in word:
            counts[char] = counts.get(char, 0) + 1

        sorted_counts = sorted(counts.values(), reverse=True)
        
        pushes = 0
        for i in range(len(sorted_counts)):
            if i < 8:
                pushes += sorted_counts[i]
            elif i < 16:
                pushes += 2 * sorted_counts[i]
            elif i < 24:
                pushes += 3 * sorted_counts[i]
            else:
                pushes += 4 * sorted_counts[i]
        
        return pushes

File: problems/3014-minimum-number-of-pushes-to-type-word-i/solution.java
import java.util.Arrays;

class Solution {
    public int minimumPushes(String word) {
        int[] freq = new int[26];
        for (char c : word.toCharArray()) {
            freq[c - 'a']++;
        }
        Arrays.sort(freq);
        int pushes = 0;
        int count = 1;
        int keys = 8;
        for (int i = 25; i >= 0; i--) {
            if (freq[i] == 0) break;
            pushes += freq[i] * count;
            keys--;
            if (keys == 0) {
                count++;
                keys = 8;
            }
        }
        return pushes;
    }
}